herc/config.py

```diff
@@ -0,0 +1,53 @@
+import pyhocon
+import threading
+import traceback
+import sys
+import importlib
+
+_config = None
```

Comment:
This file (or at least part of it) is a good candidate for being in the `__init__.py` file.  At the very least, the initialization of the config object should go there

This way, we could do something like (where in this example `config` is equivalent to your `_config`):

```python
from herc import config
config.get("scheduler.backends")
```
or: 
```python
import herc
herc.config.get("scheduler.backends")
```

The `__init__.py` could load the config when the module is first imported